app/models/members/group_member.rb

```diff
@@ -4,9 +4,20 @@ module Members
   # entity class for GroupMember
   class GroupMember < Member
     belongs_to :group, foreign_key: :namespace_id # rubocop:disable Rails/InverseOf
+    before_destroy :last_namespace_owner_member
 
     def self.sti_name
       'GroupMember'
     end
+
+    # Method to ensure we don't leave a group or project without an owner
+    def last_namespace_owner_member
+      return unless !destroyed_by_association && (namespace.owners.count == 1 && namespace.owners.last == user)
```

Comment:
Instead of checking `namespace.owners.count` you want to get all the `Owners` in the Group hierarchy, since if there is at least a `Owner` in the `self_and_ancestors` this one can be safely removed, as  long as it is not the only one.

Reply:
Whoops thanks for catching that. Updated in 20e119965680ad3aa0f927bffc00468e33c0fff7